### PR TITLE
electron-bridge: Implement electron bridge.

### PR DESCRIPTION
**Pull request by priyank-p**

### CODE REVIEW

**Comment by priyank-p** [comment]
Side-effect of force-pushing, reopening..

File: app/main/index.js [hunk 1]
```diff
@@ -224,6 +224,10 @@ app.on('ready', () => {
 	ipcMain.on('toggleAutoLauncher', (event, AutoLaunchValue) => {
 		setAutoLaunch(AutoLaunchValue);
 	});
+
+	ipcMain.on('realm-icon-changed', (event, serverURL, iconURL) => {
+		mainWindow.webContents.send('update-realm-icon', serverURL, iconURL);
```
Comment (akashnimare) [inline app/main/index.js]: `mainWindow.webContents` should be `page`.

File: app/renderer/js/electron-bridge.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+const events = require('events');
+const { ipcRenderer } = require('electron');
+
+// we have and will have some non camelcase stuff
+// while working with zulip so just turning the rule off
+// for the wole file.
```
Comment (akashnimare) [inline app/renderer/js/electron-bridge.js]: Typo - *whole

**Review by akashnimare — CHANGES_REQUESTED** [review]

**Comment by akashnimare** [comment]
This looks good. Can you fix above minor changes and resolve conflicts?

**Comment by priyank-p** [comment]
Fixed, and rebased.

**Comment by akashnimare** [comment]
@priyankp10 I'm getting following errors -

image

This doesn't let me update the icon for the first time only, later I can see the changes.

Reproduce -
* Run Zulip server
* Start Zulip app
* Add http://localhost:9991
* Try to update the realm icon

**Comment by akashnimare** [comment]
I think the bug lies in the index of the tab -
```js
const serverImgSelector = `.tab[data-tab-id="${index}"] .server-icons`;`
```
The `data-tab-id` will keep changing as you navigate in the app. Maybe get the server tab using the active class?

**Comment by priyank-p** [comment]
Yeah, using `.active` is the right solution here. Although the `data-user-id` changing is a bug we would need to fix right?

**Comment by abhigyank** [comment]
@priyankp10  Hey, Akash and me met at the Zulip retreat so we decided to complete the work on this PR so that we can make use of the `electron_bridge` feature. Thanks for implementing this! :rocket:

**Comment by priyank-p** [comment]
Done in #516.

**Pull request closed without merging**